src/cc/libclient/Writer.cc

```diff
@@ -1526,10 +1527,11 @@ class Writer::Impl :
                     return;
                 }
                 if (theStatus == kErrorReadOnly && mClosingFlag &&
-                        0 < mCloseOp.chunkId && mCloseOp.chunkVersion < 0) {
+                        0 < mCloseOp.chunkId && mKeepLeaseFlag) {
```

Comment:
Just want to confirm that you are saying we don't need the conditions to be the same. Is that correct?